jwst/photom/photom.py

```diff
@@ -134,9 +134,26 @@ def __init__(self, model, inverse=False, source_type=None, correction_pars=None)
         self.slitnum = -1
         self.specnum = -1
         self.inverse = inverse
-        self.source_type = source_type
-        if self.source_type is None and model.meta.target.source_type is not None:
-            self.source_type = model.meta.target.source_type.upper()
+        self.source_type = None
+
+        # For MultiSlitModels, only set a generic source_type value for the
+        # entire datamodel if the user has set the source_type parameter.
+        # Otherwise leave the generic source_type set to None, which will
+        # force the use of per-slit source_types later in processing.
+        if isinstance(model, datamodels.MultiSlitModel):
+            if source_type is not None:
+                self.source_type = source_type
+
+        # For non-MultiSlitModel inputs, where there's only 1 target and
+        # one source_type, use the user-provided source_type value, if it
+        # exists. Otherwise, use the generic source_type value provided in
+        # the input model (if it exists).
+        else:
+            if source_type is not None:
+                self.source_type = source_type
+            else:
+                if model.meta.target.source_type is not None:
```

Comment:
Is there a situation where `model.meta.target.source_type` could (should) be None? And if it is, will self.source_type be None as well because it hasn't been set up to this point?

Reply:
I believe that's a holdover from the days when we had to be able process groundbased test data, which was static and did not evolve to keep up with pipeline development (as we added new keywords, etc.), and hence some of those old datasets may not have `source_type` set. In those cases, suitable default values for each slit are set during the `srctype` step, so the individual slit values *should* always have a value.